src/core/job.c

```diff
@@ -557,6 +557,9 @@ static void job_log_begin_status_message(Unit *u, uint32_t job_id, JobType t) {
         if (!IN_SET(t, JOB_START, JOB_STOP, JOB_RELOAD))
                 return;
 
+        if (!unit_log_level_test(u, LOG_INFO))
+                return;
+
```

Comment:
the log_struct() call in this function further down should also use log_unit_struct(), i.e please port all calls over systematically, including those where you add an early check like above